flaps/flaps.go

```diff
@@ -42,35 +42,38 @@ type Client struct {
 }
 
 func New(ctx context.Context, app *api.AppCompact) (*Client, error) {
-	return NewWithOptions(ctx, &NewClientOpts{AppCompact: app, AppName: app.Name})
+	return newWithOptions(ctx, newClientOpts{AppCompact: app, AppName: app.Name})
 }
 
 func NewFromAppName(ctx context.Context, appName string) (*Client, error) {
-	return NewWithOptions(ctx, &NewClientOpts{AppName: appName})
+	return newWithOptions(ctx, newClientOpts{AppName: appName})
 }
 
-type NewClientOpts struct {
-	AppName    string
+type newClientOpts struct {
+	// required:
+	AppName string
+
+	// optional, avoids API roundtrip when connecting to flaps by wireguard:
 	AppCompact *api.AppCompact
-	Logger     api.Logger
+
+	// optional:
+	Logger api.Logger
 }
 
-func NewWithOptions(ctx context.Context, opts *NewClientOpts) (*Client, error) {
-	app := opts.AppCompact
-	appName := opts.AppName
-	if app != nil {
-		appName = app.Name
-	}
+func newWithOptions(ctx context.Context, opts newClientOpts) (*Client, error) {
```

Comment:
this function was public because of reasons https://github.com/superfly/flyctl/pull/2176